src/http/AuthContext.js

```diff
@@ -26,6 +38,17 @@ class AuthContext {
     get onBehalfOf() {
         return this._onBehalfOf;
     }
+
+    /**
+     * Gets the key level and resets it to low
+     *
+     * @return {string} keyLevel - key level being used
+     */
+    get keyLevel() {
```

Comment:
a property-getter that changes the value of that property seems likely to confuse us later.

what do you think of instead having methods

// Sets key level to use signing the next auth header (will revert back to LOW afterwards)
setNextKeyLevel(keyLevel)

// Get key level to use signing the next auth header (and revert value back to LOW)
getNextKeyLevel()

Reply:
I agree with larry. Getter should not modify the object.

Reply:
moved the logic to AuthHeader._keyLevel(context)